### ISSUE TITLE
mocha + ts-node hangs with node 16

### ISSUE THREAD

**Issue by cvlmtg** [TypeStrong/ts-node#1462]
I've setup mocha to run some TS tests with ts-node and under node 14 everything runs smoothly, but under node 16 it executes the tests and then hangs forever, so I have to press ctrl-c to get back to the shell prompt.

### Search Terms

- node 16
- node16
- mocha

### Expected Behavior

mocha completes its operation under node 16 as it does with node 14

### Actual Behavior

when run under node 16 all tests are executed but then mocha hangs forever

<img width="358" alt="Schermata 2021-09-16 alle 15 23 49" src="https://user-images.githubusercontent.com/6401008/133620162-dcb7e1b9-ce78-4638-8eb3-fa5c9e8beb1e.png">

<img width="352" alt="Schermata 2021-09-16 alle 15 24 11" src="https://user-images.githubusercontent.com/6401008/133620198-74a2387a-7af3-449f-b361-5c39cf1600dd.png">

### Steps to reproduce the problem

just run `yarn test`

### Minimal reproduction

https://github.com/TypeStrong/ts-node-repros/pull/18

### Specifications

ts-node v10.2.1
node v16.9.1
compiler v4.4.3

```
{
  "include": [ "src", "types" ],
  "compilerOptions": {
    "module": "esnext",
    "lib": [ "dom", "esnext" ],
    "importHelpers": true,
    "declaration": true,
    "sourceMap": true,
    "rootDir": "./src",
    "strict": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "moduleResolution": "node",
    "jsx": "react",
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true
  },
  "ts-node": {
    "compilerOptions": {
      "module": "commonjs"
    },
    "transpileOnly": true
  }
}
```

* Operating system and version: macOS 11.5.2, iTerm2 3.4.8

**Comment by cspotcode** [comment]
@cvlmtg first order of business will be to determine if this is a mocha issue or not.  Have you done any investigation to figure out if mocha bas a bug related to require hooks or other things that ts-node needs to do?

**Comment by cspotcode** [comment]
I see your reproduction contains some dependencies that are not mentioned in your description.  Have you tried removing them?  The reproduction will need to be minimal, which means including only the absolute bare minimum of dependencies.

**Comment by cvlmtg** [comment]
@cspotcode I tried to search anything related to mocha and ts-node and node 16 and found nothing.

anyway, I tried to make the repro as minimal as possible and everything works fine. If I reintroduce react-testing-library (and jsdom) the problem appears again. so (if I did everything correctly) it seems it's not a problem of ts-node but of one of those libraries. thanks for the help!

(I pushed my tests anyway, just in case they might be useful to someone)
